### PR TITLE
EP11 FIPS-session support

### PR DESCRIPTION
**Pull request by ifranzki**

### CODE REVIEW

File: usr/lib/ep11_stdll/ep11_specific.c [hunk 1]
```diff
@@ -2697,95 +2772,73 @@ static CK_RV make_maced_spki(STDLL_TokData_t *tokdata, SESSION * sess,
...
-            rc = add_to_attribute_array(&p_attrs, &attrs_len, attr->type,
-                                        attr->pValue, attr->ulValueLen);
-            if (rc != CKR_OK) {
-                TRACE_ERROR("%s adding attribute failed type=0x%lx rc=0x%lx\n",
-                            __func__, attr->type, rc);
-                goto make_maced_spki_end;
-            }
-            break;
+        is_private = ep11_is_private_object(pub_key_attrs, pub_key_attrs_len);
+        is_session = !ep11_is_session_object(pub_key_attrs, pub_key_attrs_len);
```
Comment (jschmidb) [inline usr/lib/ep11_stdll/ep11_specific.c:2973]: Just to double-check: is_session = !ep11_is_session_object ? Why the negation?

Comment (ifranzki) [reply usr/lib/ep11_stdll/ep11_specific.c:2973]: hmm good finding!  I guess the negation is wrong!

File: usr/lib/ep11_stdll/ep11_login.c [hunk 2]
```diff
@@ -0,0 +1,1232 @@
...
+    }
+
+    return rc;
+}
+
+static void increment_tcounter(CK_BYTE tcounter[XCP_ADMCTR_BYTES])
+{
+    int i;
+
+    for(i = XCP_ADMCTR_BYTES - 1; i >= 0; i--) {
```
Comment (jschmidb) [inline usr/lib/ep11_stdll/ep11_login.c:185]: space after 'for' ?

File: usr/lib/ep11_stdll/ep11_session.c [hunk 3]
```diff
@@ -351,6 +476,108 @@ static CK_RV get_pin(STDLL_TokData_t *tokdata, SESSION *session,
...
+static void remove_serial_number(STDLL_TokData_t *tokdata,
+                                 ep11_session_t *ep11_session,
+                                 ep11_serialno_t serial_number)
+{
+    CK_ULONG i;
+    CK_BOOL found = FALSE;
+
+    UNUSED(tokdata);
+
+    for(i = 0; i < ep11_session->num_serial_numbers; i++) {
```
Comment (jschmidb) [inline usr/lib/ep11_stdll/ep11_session.c:547]: space after 'for' ?

File: usr/lib/ep11_stdll/ep11_login.c (hunk 2 above)
Comment (ifranzki) [reply usr/lib/ep11_stdll/ep11_login.c:185]: fixed

**Review by jschmidb — APPROVED** [review]
lgtm

**Pull request merged**